submissions/zalevskyi/a-tiny-js-world/index.js

```diff
@@ -0,0 +1,85 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/zalevskyi/kottans-frontend/tree/main/practice/a-tiny-js-world
+   Web app: https://zalevskyi.github.io/kottans-frontend/practice/a-tiny-js-world/index.html
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const human = {species: 'human', legs: 2, hands: 2, saying: 'Bla!'}
+const cat_woman = Object.assign({}, human, {species: 'cat-woman', saying: 'Meow'})
+const pet = {legs: 4, hands: undefined}
+const dog = Object.assign({}, pet, {species: 'dog', saying: 'Woof!'})
+const cat = Object.assign({}, pet, {species: 'cat', saying: 'Meow'})
+const species = {human: human, dog: dog, cat: cat, cat_woman: cat_woman}
+const names = {
+   human: {
+      male: ['Liam', 'Noah', 'Oliver', 'William', 'Elijah', 'James', 'Benjamin', 'Lucas', 'Mason', 'Ethan'],
+      female: ['Olivia', 'Emma', 'Ava', 'Sophia', 'Isabella', 'Charlotte', 'Amelia', 'Mia', 'Evelyn', 'Abigail']
+   },
+   dog: {
+      male: ['Charlie', 'Max', 'Buddy', 'Oscar', 'Milo', 'Archie', 'Ollie', 'Toby', 'Jack', 'Teddy'],
+      female: ['Bella', 'Molly', 'Coco', 'Ruby', 'Lucy', 'Bailey', 'Daisy', 'Rosie', 'Lola', 'Frankie']
+   },
+   cat: {
+      male: ['Oliver', 'Leo', 'Milo', 'Charlie', 'Max', 'Jack', 'Simba', 'Loki', 'Oscar', 'Jasper'],
+      female: ['Luna', 'Bella', 'Lily', 'Lucy', 'Kitty', 'Callie', 'Nala', 'Zoe', 'Chloe', 'Sophie']
+   },
+   cat_woman: {
+      female: ['Daisy', 'Stella', 'Cleo']
+   }
+}
+const inhabitans = []
+for (let kind in names) {
+   for (let gender in names[kind]) {
+      inhabitans.push(...names[kind][gender].map(name => Object.assign({},species[kind],{gender: gender, name: name})))
+   }
+}
+inhabitans.forEach((obj, index, arr) => {
+   if (obj.species==='human') Object.defineProperty(arr[index], 'saying', {get: () => `Hi! I'm ${obj.name}`})
+   if (obj.species==='cat-woman') Object.defineProperty(arr[index], 'saying', {get: () => cat.saying})
+})
```

Comment:
Let's put the fragments above into well named functions and call those immediately.
Seems like an overkill here (as each gonna be called only once) but will add to readability a lot.